server/redux/reducers/user-reducer.js

```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
 
 /* --------------- THUNK ACTION CREATORS --------------- */
 
-const createAndEmitUser = socket => {
+const createAndEmitUser = (socket, user) => {
   return dispatch => {
     const userId = socket.id;
-    const user = Map(createUser(userId));
-    dispatch(addUser(user));
-    socket.on('sceneLoad', () => {
-      socket.emit('createUser', user);
-    });
+    const displayName = user.displayName;
+    const newUser = Map(createUser(userId, displayName));
+    dispatch(addUser(newUser));
+    console.log('User created');
+    if (socket.sceneLoaded) {
+      socket.emit('createUser', newUser);
```

Comment:
At this point, it seems like maybe this 'createUser' event should be 'renderAvatar' or something more semantically appropriate. It's way to similar to the addUser and createUser functions.